projects/ion/src/lib/switch/switch.component.spec.ts

```diff
@@ -0,0 +1,129 @@
+import { Component } from '@angular/core';
+import {
+  FormControl,
+  FormGroup,
+  FormsModule,
+  ReactiveFormsModule,
+} from '@angular/forms';
+import { fireEvent, render, screen } from '@testing-library/angular';
+
+import { SafeAny } from '../utils/safe-any';
+import { IonSwitchComponent } from './switch.component';
+import { SwitchSize } from './types';
+import { CommonModule } from '@angular/common';
+
+let ionSwitch: HTMLElement;
+
+const emitValue = {
+  emit: jest.fn(),
+} as SafeAny;
+
+const sizes: SwitchSize[] = ['sm', 'md', 'lg'];
+
+const sut = async (
+  customProps: Partial<IonSwitchComponent> = {}
+): Promise<HTMLElement> => {
+  await render(IonSwitchComponent, {
+    componentProperties: { ...customProps, atValueChange: emitValue },
+  });
+  return screen.getByTestId('ion-switch');
+};
+
+describe('IonSwitchComponent', () => {
+  describe('General', () => {
+    beforeEach(async () => {
+      ionSwitch = await sut();
+    });
+    it('should render switch', async () => {
+      expect(ionSwitch).toBeInTheDocument();
+    });
+    it('should render switch with default class', () => {
+      expect(ionSwitch).toHaveClass('ion-switch');
+    });
+    it('should change class to active when switch is clicked', () => {
+      fireEvent.click(ionSwitch);
+      expect(ionSwitch).toHaveClass('ion-switch--active');
+    });
+    it('should remove active class when switch is clicked twice', () => {
+      fireEvent.click(ionSwitch);
+      expect(ionSwitch).toHaveClass('ion-switch--active');
+
+      fireEvent.click(ionSwitch);
+      expect(ionSwitch).toHaveClass('ion-switch');
+    });
+    it('should emit correct value when switch is clicked', () => {
+      fireEvent.click(ionSwitch);
+      expect(emitValue.emit).toHaveBeenCalledWith(true);
+
+      fireEvent.click(ionSwitch);
+      expect(emitValue.emit).toHaveBeenCalledWith(false);
+    });
+  });
+  describe.each(sizes)('Sizes - %s', size => {
+    beforeEach(async () => {
+      ionSwitch = await sut({ size });
+    });
+    it(`should render a switch with size attribute and value '${size}'`, () => {
+      expect(ionSwitch).toHaveAttribute('size', size);
+    });
+  });
+  describe('Disabled', () => {
+    it('should be enabled by default', async () => {
+      ionSwitch = await sut();
+      expect(ionSwitch).not.toBeDisabled();
+    });
+    it('should be disabled', async () => {
+      ionSwitch = await sut({ disabled: true });
+      expect(ionSwitch).toBeDisabled();
+    });
+  });
+});
+
+@Component({
+  template: `
+    <form [formGroup]="formGroup">
+      <ion-switch formControlName="name" key="name"></ion-switch>
```

Comment:
This optional can be call like: <ion-switch formControlName="name" key="name"/>. It's shorter.